mitmproxy/filt.py

```diff
@@ -155,9 +155,9 @@ class FHead(_Rex):
     flags = re.MULTILINE
 
     def __call__(self, f):
-        if f.request and self.re.search(bytes(f.request.headers)):
+        if f.request and self.re.search(bytes(f.request.headers).decode()):
```

Comment:
Okay, so I've removed `filt` related code from here! I think this can be merged now.